<?xml version="1.0" encoding="UTF-8" ?>
<testsuite tests="55" failures="2" name="phones.secret.GeneralFeaturesTest" time="0.131" errors="0" skipped="46">
  <properties>
    <property name="java.runtime.name" value="Java(TM) SE Runtime Environment"/>
  </properties>
  <testcase classname="phones.secret.GeneralFeaturesTest" name="when_calling_and_call_incoming__phone_continues_calling_first_target" time="0.015">
    <skipped/>
  </testcase>
  <testcase classname="phones.secret.GeneralFeaturesTest" name="a_new_phone_is_idle" time="0.001"/>
  <testcase classname="phones.secret.GeneralFeaturesTest" name="after_dialing__press_green_to_call" time="0.008">
    <failure message="When Alice dials &apos;555-THE-BOBSTER&apos;
 and Alice presses green,
Expected: Alice status is &lt;CALLING&gt;
     but: was &lt;IDLE&gt;" type="java.lang.AssertionError">java.lang.AssertionError: When Alice dials &apos;555-THE-BOBSTER&apos;
 and Alice presses green,
Expected: Alice status is &lt;CALLING&gt;
     but: was &lt;IDLE&gt;
	at phones.secret.FeatureTestBase$2.mismatch(FeatureTestBase.java:358)
	at org.cthul.matchers.fluent.builder.FluentAssertBuilder._apply(FluentAssertBuilder.java:164)
	at org.cthul.matchers.fluent.builder.FluentAssertBuilder._apply(FluentAssertBuilder.java:24)
	at org.cthul.matchers.fluent.builder.AbstractFluentStepBuilder$1.apply(AbstractFluentStepBuilder.java:115)
	at org.cthul.matchers.fluent.ext.Extensions$BasicAdapter$1$1.apply(Extensions.java:95)
	at org.cthul.matchers.fluent.ext.Extensions$BasicAdapter$1$1.apply(Extensions.java:95)
	at org.cthul.matchers.fluent.builder.FluentStepBuilder._apply(FluentStepBuilder.java:29)
	at org.cthul.matchers.fluent.builder.AbstractFluentStepBuilder._match(AbstractFluentStepBuilder.java:62)
	at org.cthul.matchers.fluent.builder.AbstractFluentStepBuilder._match(AbstractFluentStepBuilder.java:69)
	at org.cthul.matchers.fluent.builder.AbstractFluentStepBuilder.is(AbstractFluentStepBuilder.java:216)
	at phones.secret.GeneralFeaturesTest.after_dialing__press_green_to_call(GeneralFeaturesTest.java:64)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at phones.secret.FeatureTestBase$ExceptionContextRule$1.evaluate(FeatureTestBase.java:390)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
</failure>
  </testcase>
  <testcase classname="phones.secret.GeneralFeaturesTest" name="after_dialing__target_is_idle" time="0">
    <error message="When Alice sets forwarding to &apos;555-THE-BOBSTER&apos;,
Got UnsupportedOperationException: Redirect" type="phones.secret.FeatureTestBase$ExceptionContextRule$1$1ContextException">phones.secret.FeatureTestBase$ExceptionContextRule$1$1ContextException: When Alice sets forwarding to &apos;555-THE-BOBSTER&apos;,
Got UnsupportedOperationException: Redirect
java.lang.UnsupportedOperationException: Redirect
	at phones.Phone.setForward(Phone.java:30)
	at phones.secret.FeatureTestBase$WhenPhone.setForward(FeatureTestBase.java:259)
	at phones.secret.ForwardingFeatureTest.forwarding_support(ForwardingFeatureTest.java:14)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at phones.secret.FeatureTestBase$ExceptionContextRule$1.evaluate(FeatureTestBase.java:390)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
</error>
  </testcase>
</testsuite>